cupy/math/sumprod.py

```diff
@@ -192,6 +192,65 @@ def cumprod(a, axis=None, dtype=None, out=None):
     """
     return _cum_core(a, axis, dtype, out, _cumprod_kern, _cumprod_batch_kern)
 
+def diff(a, n=1, axis=-1):
+  """Calculate the n-th discrete difference along the given axis.
+  Args:
+      a (cupy.ndarray): Input array.
+      n (int): The number of times values are differenced. If zero, the input
+                  is returned as-is.
+      axis (int): The axis along which the difference is taken, default is the
+                    last axis.
+  Returns:
+      cupy.ndarray: The result array.
+  .. seealso:: :func:`numpy.diff`
+  """
+  
+  if n == 0:
+      return a
+  if n < 0:
+      raise ValueError(
+          "order must be non-negative but got " + repr(n))
+
+  a = cupy.asanyarray(a)
+  nd = a.ndim
+  
+  slice1 = [slice(None)] * nd
+  slice2 = [slice(None)] * nd
+  slice1[axis] = slice(1, None)
+  slice2[axis] = slice(None, -1)
+  slice1 = tuple(slice1)
+  slice2 = tuple(slice2)
+
+  op = not_equal if a.dtype == numpy.bool_ else cupy.subtract
+  for _ in range(n):
+      a = op(a[slice1], a[slice2])
+
+  return a
+
+def unwrap(p, discont=numpy.pi, axis=-1):
```

Comment:
Please send another PR and move to `cupy/math/trigonometric.py` line 102.
There are `# TODO(okuta): Implement unwrap`.